community/detectors/mlflow_cve_2023_6014/src/main/java/com/google/tsunami/plugins/detectors/cves/cve20236014/Cve20236014VulnDetector.java

```diff
@@ -0,0 +1,177 @@
+  @VisibleForTesting static final String DETECTION_STRING = "Successfully signed up user:";
+
+  @VisibleForTesting
+  static final String VULN_DESCRIPTION =
+      "On MLflow versions prior to 2.8.0, unauthenticated users are able to arbitrarily create an"
+          + " account in MLflow server or MLflow UI by bypassing any authentication requirement.\n";
+
+  private final HttpClient httpClient;
+  private final Clock utcClock;
+
+  private static final String ALPHA_NUMERIC_STRING = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+  private static final SecureRandom secureRandom = new SecureRandom();
+
+  public static String GenerateRandomString(int length) {
+    StringBuilder builder = new StringBuilder();
+    for (int i = 0; i < length; i++) {
+      int index = secureRandom.nextInt(ALPHA_NUMERIC_STRING.length());
+      builder.append(ALPHA_NUMERIC_STRING.charAt(index));
+    }
+    return builder.toString();
+  }
+
+  @Inject
+  Cve20236014VulnDetector(@UtcClock Clock utcClock, HttpClient httpClient) {
+    this.httpClient = checkNotNull(httpClient);
+    this.utcClock = checkNotNull(utcClock);
+  }
+
+  private static boolean isWebServiceOrUnknownService(NetworkService networkService) {
+    return networkService.getServiceName().isEmpty()
+        || NetworkServiceUtils.isWebService(networkService)
+        || NetworkServiceUtils.getServiceName(networkService).equals("unknown");
+  }
+
+  private static StringBuilder buildTarget(NetworkService networkService) {
+    StringBuilder targetUrlBuilder = new StringBuilder();
+    if (NetworkServiceUtils.isWebService(networkService)) {
+      targetUrlBuilder.append(NetworkServiceUtils.buildWebApplicationRootUrl(networkService));
+    } else {
+      targetUrlBuilder
+          .append("https://")
+          .append(toUriAuthority(networkService.getNetworkEndpoint()))
+          .append("/");
+    }
+    return targetUrlBuilder;
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("CVE-2023-6014 starts detecting.");
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(Cve20236014VulnDetector::isWebServiceOrUnknownService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    String targetVulnerabilityUrl = buildTarget(networkService).append(VUL_PATH).toString();
+    try {
+      String randomUsername = GenerateRandomString(10);
+      String randomPassword = GenerateRandomString(10);
+      logger.atInfo().log(
+          "Attempting to create a user (" + randomUsername + ":" + randomPassword + ")");
```

Comment:
Please use flogger's formatting logs:
```suggestion
          "Attempting to create a user (%s: %s)", randomUsername, randomPassword);
```